lib/shared.h

```diff
@@ -0,0 +1,28 @@
+#include <stdlib.h>
```

Comment:
This header is exactly the same as `marshal.h`. Why is it needed? Seems like it's just adding clutter.